packages/autocomplete-plugin-recent-searches/src/getTemplates.tsx

```diff
@@ -0,0 +1,55 @@
+/** @jsx createElement */
+import { reverseHighlightHit, SourceTemplates } from '@algolia/autocomplete-js';
+
+import { RecentSearchesItem } from './types';
+
+export type GetTemplatesParams = {
+  onRemove(id: string): void;
+};
+
+export function getTemplates<TItem extends RecentSearchesItem>({
+  onRemove,
+}: GetTemplatesParams): SourceTemplates<TItem> {
+  return {
+    item({ item, createElement, Fragment }) {
+      return (
+        <Fragment>
+          <div className="aa-ItemIcon aa-ItemIcon--no-border">
+            <svg width="20" height="20" viewBox="0 0 22 22" fill="currentColor">
+              <path d="M0 0h24v24H0z" fill="none" />
+              <path d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9zm-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8H12z" />
+            </svg>
+          </div>
+          <div className="aa-ItemContent">
+            <div className="aa-ItemContentTitle">
+              {reverseHighlightHit<any>({
```

Comment:
We should have a type for this? Like in `querySuggestions`

Reply:
`reverseHighlightHit` (and all highlight utils) expect an Algolia hit and we consider a hit to have an `objectID` key. A recent item doesn't have an `objectID`, but it does have the Algolia properties likes `_highlightResult`. That's why we cast it as `any` here.

We might want to reconsider the `objectID` check and rather consider an Algolia hit as long as it has a `_highlightResult` or `_snippetResult` key?